src/m_services.c

```diff
@@ -1108,6 +1178,261 @@ int m_sjr(aClient *cptr, aClient *sptr, int parc, char *parv[], AliasInfo *ai)
+                {
+                    sendto_one(sptr, ":%s PRIVMSG %s :CONF STATUS %s %ld %ld %s", me.name, parv[0], readonly_files[cnt], sb.st_size, sb.st_mtime, md5file(readonly_files[cnt]));
+                }
+            }
+        }
+        else if(!mycmp(parv[3], "STATUS"))
+        {
+            if(!(tmp = is_allowed_configfile(parv[4], 0)))
+            {
+                sendto_realops_lev(DEBUG_LEV, "SVSCTRL CONF STATUS Error: Access denied to %s", parv[4]);
+                sendto_one(sptr, ":%s PRIVMSG %s :CONF STATUS %s ERROR :Access Denied", me.name, parv[0], parv[4]);
+                return 0;
+            }
+            strcpy(fn, tmp);
+            if(stat(fn, &sb) == -1)
+            {
+                sendto_realops_lev(DEBUG_LEV, "SVSCTRL CONF STATUS Error: Couldn't stat %s", fn);
+                sendto_one(sptr, ":%s PRIVMSG %s :CONF STATUS %s ERROR :Couldn't stat file", me.name, parv[0], fn);
+                return 0;
+            }
+            sendto_one(sptr, ":%s PRIVMSG %s :CONF STATUS %s %ld %ld %s", me.name, parv[0], fn, sb.st_size, sb.st_mtime, md5file(fn));
+        }
+        else if(!mycmp(parv[3], "READ"))
+        {
+            if(!(tmp = is_allowed_configfile(parv[4], 0)))
+            {
+                sendto_realops_lev(DEBUG_LEV, "SVSCTRL CONF READ Error: Access denied to %s", parv[4]);
+                sendto_one(sptr, ":%s PRIVMSG %s :CONF READ %s ERROR :Access Denied", me.name, parv[0], parv[4]);
+                return 0;
+            }
+            strcpy(fn, tmp);
+            strcpy(fn_bak, tmp); strcat(fn_bak, ".baksvs");
+            strcpy(fn_new, tmp); strcat(fn_new, ".newsvs");
+            f = fopen(fn, "r");
+            line_counter = 0;
+            if(!f)
+            {
+                sendto_realops_lev(DEBUG_LEV, "SVSCTRL CONF READ Error: Couldn't open %s for reading", fn);
+                sendto_one(sptr, ":%s PRIVMSG %s :CONF READ %s ERROR :Couldn't open file for reading", me.name, parv[0], fn);
+                return 0;
+            }
+            if(stat(fn, &sb) == -1)
+            {
+                sendto_realops_lev(DEBUG_LEV, "SVSCTRL CONF READ Error: Couldn't stat %s", fn);
+                sendto_one(sptr, ":%s PRIVMSG %s :CONF READ %s ERROR :Couldn't stat file", me.name, parv[0], fn);
+                return 0;
+            }
+            sendto_one(sptr, ":%s PRIVMSG %s :CONF %s SOF %ld %ld", me.name, parv[0], fn, sb.st_size, sb.st_mtime);
+            while(fgets(line, BUFSIZE, f) != NULL)
+            {
+                while((tmp = strchr(line,'\r'))) *tmp = '\0';
+                while((tmp = strchr(line,'\n'))) *tmp = '\0';
+                line_counter++;
+                sendto_one(sptr, ":%s PRIVMSG %s :CONF %s %d :%s", me.name, parv[0], fn, line_counter, line);
+            }
+            sendto_one(sptr, ":%s PRIVMSG %s :CONF %s EOF %d %ld %s", me.name, parv[0], fn, line_counter, sb.st_size, md5file(fn));
```

Comment:
I think all of the replies for reading should have the "READ" parameter before the file name to be consistent with the rest of the system

:%s PRIVMSG %s :CONF READ %s SOF %ld %d instead of :%s PRIVMSG %s :CONF %s SOF %ld %ld

The same for the line replies and the EOF reply.